core/src/main/java/hudson/model/View.java

```diff
@@ -1302,20 +1294,14 @@ private static View copy(StaplerRequest req, ViewGroup owner, String name) throw
      * @param name Alternative name to use or <tt>null</tt> to keep the one in xml.
      */
     public static View createViewFromXML(String name, InputStream xml) throws IOException {
-        InputStream in = new BufferedInputStream(xml);
-        try {
+
+        try (InputStream in = new BufferedInputStream(xml)) {
             View v = (View) Jenkins.XSTREAM.fromXML(in);
             if (name != null) v.name = name;
             checkGoodName(v.name);
             return v;
-        } catch(StreamException e) {
-            throw new IOException("Unable to read",e);
-        } catch(ConversionException e) {
-            throw new IOException("Unable to read",e);
-        } catch(Error e) {// mostly reflection errors
+        } catch(StreamException|ConversionException|Error e) {// mostly reflection errors
```

Comment:
🐛 but for the original code. Errors may be non-recoverable and should not wrapped by common exceptions in such way. they should be propagated instead